lake/oneshot_cls.py

```diff
@@ -104,24 +105,46 @@ def main():
     model = CLS(image_shape, config, device=device, writer=writer).to(device)
 
   if not pretrained_model_path:
-    background_loader = torch.utils.data.DataLoader(
-        datasets.Omniglot('./data', background=True, download=True, transform=image_tfms),
-        batch_size=config['pretrain_batch_size'], shuffle=True)
+    val_split = 0.175
+
+    dataset = datasets.Omniglot('./data', background=True, download=True, transform=image_tfms)
+
+    val_size = round(val_split * len(dataset))
+    train_size = len(dataset) - val_size
+
+    train_set, val_set = torch.utils.data.random_split(dataset, [train_size, val_size])
+
+    train_loader = torch.utils.data.DataLoader(train_set, batch_size=config['pretrain_batch_size'], shuffle=True)
+    val_loader = torch.utils.data.DataLoader(val_set, batch_size=config['pretrain_batch_size'], shuffle=True)
 
     # Pre-train the model
     for epoch in range(start_epoch, config['pretrain_epochs'] + 1):
       model.train()
 
-      for batch_idx, (data, target) in enumerate(background_loader):
+      for batch_idx, (data, target) in enumerate(train_loader):
         data, target = data.to(device), target.to(device)
 
         losses, _ = model(data, labels=target if model.is_ltm_supervised() else None, mode='pretrain')
         pretrain_loss = losses['ltm']['memory']['loss'].item()
```

Comment:
It looks to me like both the train and validation loss are being logged at: `losses['ltm']['memory']['loss']` within the CLS class. It happens within `MemoryInterface` which is a parent of the LTM class.
So they will overwrite each other and there will only be one plot in Tensorboard.
Is that correct?

Reply:
It shouldn't override it because of the 'mode'. The 'training' loss will be prefixed by `pretrain` (the mode) and then the validation loss should be prefixed by `validate`